src/providers/mssql/qgsmssqlconnection.cpp

```diff
@@ -436,15 +436,15 @@ QString QgsMssqlConnection::buildQueryForTables( bool allowTablesWithNoGeometry,
   {
     query += QLatin1String( "f_table_schema, f_table_name, f_geometry_column, srid, geometry_type, 0, coord_dimension FROM geometry_columns" );
     if ( !notSelectedSchemas.isEmpty() )
-      query += QStringLiteral( " WHERE f_table_schema NOT IN %1" ).arg( notSelectedSchemas );
+      query += QStringLiteral( " WHERE QUOTENAME(f_table_schema) NOT IN %1" ).arg( notSelectedSchemas );
```

Comment:
Is this correct? To me it looks like we're comparing a quoted identifier (from QUOTENAME) vs a quoted value (from QgsMssqlProvider::quotedValue) in notSelectedSchemas

Reply:
The QUOTENAME should be right, because its applied to both sides of the comparison:

```sql
QUOTENAME(f_table_schema) NOT IN (QUOTENAME('ANGLO\User'))
```

becomes at runtime:
```sql 
[actual_schema_name] NOT IN ([ANGLO\User])
```

Both sides are converted to bracketed identifiers by SQL Server before comparison, ensuring its comparing like with like. Its letting SQL Server handle escaping and format consistency, rather than trying to manage string literals vs identifiers ourselves.

Reply:
You're right, won't wokr... cancelled